pinot-core/src/main/java/org/apache/pinot/core/operator/transform/transformer/datetime/BaseDateTimeTransformer.java

```diff
@@ -42,14 +46,26 @@ public abstract class BaseDateTimeTransformer<I, O> implements DataTransformer<I
   private final DateTimeFormatter _outputDateTimeFormatter;
   private final DateTimeGranularitySpec _outputGranularity;
   private final long _outputGranularityMillis;
-  private final SDFDateTimeTruncate _dateTimeTruncate;
+  private final SDFDateTimeTruncate _sdfDateTimeTruncate;
+  private final MillisDateTimeTruncate _millisDateTimeTruncate;
+  private final Chronology _bucketingChronology;
+  private final MutableDateTime _dateTime;
+  private final StringBuilder _printBuffer;
 
   private interface SDFDateTimeTruncate {
-    String truncate(DateTime dateTime);
+    String truncate(MutableDateTime dateTime);
   }
 
-  public BaseDateTimeTransformer(@Nonnull DateTimeFormatSpec inputFormat, @Nonnull DateTimeFormatSpec outputFormat,
-      @Nonnull DateTimeGranularitySpec outputGranularity) {
+  private interface MillisDateTimeTruncate {
+    long truncate(MutableDateTime dateTime);
+  }
+
+  public BaseDateTimeTransformer(
+      @Nonnull DateTimeFormatSpec inputFormat,
```

Comment:
(Convention) We usually only annotate `Nullable` and treat others as `Nonnull`

Reply:
Ok. These parameters were @Nonnull in master already. Should I get rid of the annotations ?

Reply:
Removed.